packages/asl-pages/pages/project-version/middleware/diff/compare.js

```diff
@@ -6,7 +6,19 @@ const parseValue = (val) => {
   if (typeof val === 'string') {
     val = JSON.parse(val || '{}');
   }
-  return Value.fromJSON(val || {});
+  return getText(val.document.nodes);
+};
+
+const getText = (nodes) => {
+  let tempText = '';
+  nodes?.forEach((element) => {
+    if (element?.object === 'block') {
+      tempText += getText(element?.nodes);
+    } else if (element?.object === 'text') {
+      tempText += element.text + ' ';
+    }
+  });
+  return tempText;
```

Comment:
Is it necessary that this includes an extra space at the end of the list? If not consider using `join`

```suggestion
const getText = (nodes = []) => {
  return nodes.flatMap(
    node => {
      if (node?.object === 'block') {
        return getText(node?.nodes);
      } else if (node?.object === 'text') {
        return [element.text];
      } else {
        return [];
      }
    }
  )
  .join(' ');
```

Also this is the same as the method in `packages/asl-projects/client/helpers/document-diff.js` can they be refactored into a single helper function.